pkg/git/git.go

```diff
@@ -0,0 +1,149 @@
+package git
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/treeverse/lakefs/pkg/ioutils"
+	"golang.org/x/exp/slices"
+)
+
+const (
+	IgnoreFile        = ".gitignore"
+	IgnoreDefaultMode = 0644
+)
+
+func git(dir string, args ...string) (int, string) {
+	stdout := new(strings.Builder)
+	stderr := new(strings.Builder)
+	cmd := exec.Command("git", args...)
+	cmd.Dir = dir
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+	err := cmd.Run()
+	if err != nil {
+		var exitError *exec.ExitError
+		if errors.As(err, &exitError) {
+			return exitError.ExitCode(), stderr.String()
+		}
+		return -1, err.Error()
+	}
+	return 0, stdout.String()
+}
+
+// IsGitRepository Return true if dir is a path to a directory in a git repository, false otherwise
+func IsGitRepository(dir string) bool {
+	rc, _ := git(dir, "rev-parse", "--is-inside-work-tree")
+	return rc == 0
+}
+
+// GetGitRepositoryPath Returns the git repository root path if dir is a directory inside a git repository, otherwise returns error
+func GetGitRepositoryPath(dir string) (string, error) {
+	rc, out := git(dir, "rev-parse", "--show-toplevel")
+	if rc == 0 {
+		return strings.TrimSpace(out), nil
+	}
+	if strings.Contains(out, ErrNotARepository.Error()) {
+		return "", ErrNotARepository
+	}
+	return "", fmt.Errorf("%w: exit code %d: %s", ErrGitError, rc, out)
+}
+
+// Ignore Modify .ignore file to include a section headed by the marker string and contains the provided paths
+// If section exists, it will append paths to the given section, otherwise writes the sections at the end of the file.
+// File paths must be absolute.
+// Creates the .ignore file if it doesn't exist.
+func Ignore(dir string, ignorePaths, excludePaths []string, marker string) (string, error) {
```

Comment:
general note - if we use io/fs (passing fs.FS to this call) we can abstract the os level and have the code test better without doing actual os level calls.

Reply:
I don't think we can pass an FS object since the dir passed might not be the git repository root